sharpy/knowledges/knowledge.py

```diff
@@ -135,6 +141,15 @@ def pre_start(self, ai: "KnowledgeBot", additional_managers: Optional[List[Manag
 
         self.my_worker_type = self.unit_values.get_worker_type(self.my_race)
 
+    # noinspection PyAttributeOutsideInit
+
+    @property
+    def managers(self) -> List[ManagerBase]:
+        """
+        Returns all registered managers.
+        """
+        return list(self._managers_dict.values())
```

Comment:
Creating a new list each time is less than ideal. Cache it, or perhaps values() can already be iterated upon.